### PR TITLE
Update package.json

### PR DESCRIPTION
**Pull request by cortopy**
Add missing dependency of yargs

#### What it does
Fixes missing dependency when installing

### CODE REVIEW

File: dev-packages/cli/package.json [hunk 1]
```diff
@@ -31,6 +31,7 @@
   },
   "dependencies": {
     "@theia/application-manager": "^0.11.0",
-    "@theia/application-package": "^0.11.0"
+    "@theia/application-package": "^0.11.0",
+    "yargs": "^14.2.0"
```
Comment (akosyakov) [inline dev-packages/cli/package.json]: dev dependencies should be declare in the root package

**Comment by akosyakov** [comment]
We have it already from yarn workspace hoisting: https://github.com/eclipse-theia/theia/blob/aff11bec2a0424a63e506613d9e23f2e6bdd1d42/packages/core/package.json#L47 Why do we need to manage it in another place as well?

**Comment by cortopy** [comment]
@akosyakov I installed `@theia/cli` from the npm registry so scoping in the monorepo is lost. As a result, yarns dependency wasn't installed. Looking at the dependencies of `@theia/cli`, I can only see two and neither of those depends on `@theia/core`

I first installed @theia packages with `pnpm`, and then I realised that the monorepo as a whole relies on the risky flat structure that comes with `yarn`.

Switching to `yarn` fixed it, but in my opinion this is a bad practice as it relies on users also installing `@theia/core` without this dependency having been declared anywhere in the dependency tree of `@theia/cli`

**Comment by cortopy** [comment]
@vince-fugnitto I've already signed it, but for some reason it's failed. I think it's because I use a `noreply` address in github. Any suggestions as to how to proceed?

**Comment by akosyakov** [comment]
@cortopy oh, sorry, I thought it is added as a dev dependency, for runtime it totally makes sense!

**Comment by kittaakos** [comment]
> Add missing dependency of yargs

Please add `"yargs": "^11.1.0"`. So we make sure we have the same `yargs` version all over the application.

> Any suggestions as to how to proceed?
> I've already signed it,

No you did not, please see your commit message here: https://github.com/eclipse-theia/theia/pull/6418/commits/6b326e47d04f614db0d2a5206f1850eabe487c08
You can try `git commit --amend --author="Your Name <youremail@domain.com>"` if it is a GH private email issue.

Also, make sure you commit your `yarn.lock` changes. Otherwise, the new dependency in the `package.json` is half-complete fix.

**Comment by cortopy** [comment]
@kittaakos 

> Please add "yargs": "^11.1.0". 

Done

> Also, make sure you commit your yarn.lock changes. Otherwise, the new dependency in the package.json is half-complete fix.

I changed the version and run an install, but the `yarn.lock` didn't seem to change according to git, I guess because the dep is already in the tree?

> No you did not, please see your commit message here: 6b326e4
You can try git commit --amend --author="Your Name <youremail@domain.com>" if it is a GH private email issue.

I did sign the agreement last Saturday using my github account. This is what I see if I come back to it. I don't know what else I need to do. I do my commits with my `noreply` address for privacy reasons

Screenshot 2019-10-22 at 17 06 04

**Comment by kittaakos** [comment]
> Done

👍 

> the dep is already in the tree?

Correct. This is expected.

> sign the agreement last Saturday using my github account

You have to sign the commit too: https://git-scm.com/docs/git-commit#Documentation/git-commit.txt---signoff

Your commit should look like this. See the last line in the message:

Screen Shot 2019-10-23 at 09 09 11

Yours is:
Screen Shot 2019-10-23 at 09 12 23


Please make sure to use the same email address for the commit signing as you did use for the ECA registration.

**Comment by cortopy** [comment]
@kittaakos sorry but I'm struggling with this

I've just signed off the commit. This is what I see now
Screenshot 2019-10-23 at 16 29 15

But it's still not working. I've tried to change the email address for my Eclipe account to the one I signed off with but it says I need to validate it first. Since it's a noreply email address, I don't have the means to do so

**Comment by vince-fugnitto** [comment]
> But it's still not working. I've tried to change the email address for my Eclipse account to the one I signed off with but it says I need to validate it first. Since it's a noreply email address, I don't have the means to do so.

@cortopy 
If you'd like, I can open a pull request with you as the co-author in order to land the changes.
I don't believe the ECA likely supports `noreply` email addresses and if I can understand if you'd wish to keep it that way for privacy concerns.

**Comment by cortopy** [comment]
@vince-fugnitto I think that's what it is. I was already getting an error message about that when I signed the agreement on Saturday.

I've manage to approve the email change (I suspect I won't be getting emails from Eclipse any more) but it's still failing

I'd be very happy to sign any other PR as co-author if that's any easier

**Comment by vince-fugnitto** [comment]
@cortopy I have the PR here https://github.com/eclipse-theia/theia/pull/6443, are you fine with it?

**Comment by cortopy** [comment]
@vince-fugnitto looks awesome. Thanks so much!!

**Comment by akosyakov** [comment]
That's really a pity that we lost one new contributor because of ECA sign-off 😢

**Pull request closed without merging**